detection_rules/devtools.py

```diff
@@ -1279,17 +1281,44 @@ def update_rule_data_schemas():
         cls.save_schema()
 
 
-@schemas_group.command("generate-endgame")
+@schemas_group.command("generate")
 @click.option("--token", required=True, prompt=get_github_token() is None, default=get_github_token(),
               help="GitHub token to use for the PR", hide_input=True)
-@click.option("--endgame-version", "-e", required=True, help="Tagged version from TBD. e.g., 1.9.0")
+@click.option("--schema", "-s", required=True, type=click.Choice(["endgame", "ecs", "beats", "endpoint"]),
+              help="Schema to generate")
+@click.option("--schema-version", "-sv", help="Tagged version from TBD. e.g., 1.9.0")
+@click.option("--endpoint-target", "-t", type=str, default="endpoint", help="Target endpoint schema")
```

Comment:
What about the [endgame custom schema](https://github.com/elastic/endpoint-package/blob/main/custom_schemas/endgame/custom_endgame.yaml)

Reply:
The goal is provide rule authors with the capability to validate on fields that are not core in ECS but generated in docs such as `process.name.caseless`, while not inflating how many schemas we manage in the repo. At the moment there is only a single rule using `process.name.caseless` so only the `process` schema was downloaded. This is why endgame was not downloaded.

Endgame's schema is managed separately with `EndgameSchemaManager`.

Reply:
Addressed this to add resiliency for the target being a folder by adjusting the url if a 404 is given during the request since the target has been validated before reaching that step.
<img width="1363" alt="Screenshot 2023-06-28 at 3 41 23 PM" src="https://github.com/elastic/detection-rules/assets/99630311/0be0a671-1918-47a9-b3d3-ff55eb4f6e6e">